components/script/dom/element.rs

```diff
@@ -3393,6 +3394,34 @@ impl<'a> SelectorsElement for DomRoot<Element> {
             }
         }
     }
+
+    fn add_element_unique_hashes(&self, filter: &mut BloomFilter) -> bool {
+        let mut f = |hash| filter.insert_hash(hash & BLOOM_HASH_MASK);
+
+        // We can't use style::bloom::each_relevant_element_hash(*self, f)
+        // since DomRoot<Element> doesn't have the TElement trait.
```

Comment:
BTW, not worth it to do it in this PR, but should `DomRoot<Element>` implement TElement?

Reply:
Perhaps...Aren't we usually processing a LayoutDom node though?